src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -0,0 +1,337 @@
+    public sealed class ReleasePipelineRunner : IServiceImplementation, IReleasePipelineRunner
+    {
+        public ReleasePipelineRunner(
+            IReliableStateManager stateManager,
+            ILogger<ReleasePipelineRunner> logger,
+            BuildAssetRegistryContext context)
+        {
+            StateManager = stateManager;
+            Logger = logger;
+            Context = context;
+        }
+
+        public IReliableStateManager StateManager { get; }
+        public ILogger<ReleasePipelineRunner> Logger { get; }
+        public BuildAssetRegistryContext Context { get; }
+
+        private const string RunningPipelineDictionaryName = "runningPipelines";
+        private static HashSet<string> InProgressStatuses = new HashSet<string>() { "notstarted", "scheduled", "queued", "inprogress" };
+
+        public async Task StartAssociatedReleasePipelinesAsync(int buildId, int channelId)
+        {
+            IReliableConcurrentQueue<ReleasePipelineRunnerItem> queue =
+                await StateManager.GetOrAddAsync<IReliableConcurrentQueue<ReleasePipelineRunnerItem>>("queue");
+            using (ITransaction tx = StateManager.CreateTransaction())
+            {
+                await queue.EnqueueAsync(
+                    tx,
+                    new ReleasePipelineRunnerItem
+                    {
+                        BuildId = buildId,
+                        ChannelId = channelId
+                    });
+                await tx.CommitAsync();
+            }
+        }
+
+        public async Task RunAsync(CancellationToken cancellationToken)
+        {
+            IReliableConcurrentQueue<ReleasePipelineRunnerItem> queue =
+                await StateManager.GetOrAddAsync<IReliableConcurrentQueue<ReleasePipelineRunnerItem>>("queue");
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    using (ITransaction tx = StateManager.CreateTransaction())
+                    {
+                        ConditionalValue<ReleasePipelineRunnerItem> maybeItem = await queue.TryDequeueAsync(
+                            tx,
+                            cancellationToken);
+                        if (maybeItem.HasValue)
+                        {
+                            ReleasePipelineRunnerItem item = maybeItem.Value;
+                            using (Logger.BeginScope(
+                                "Triggering release pipelines associated with channel {channelId} for build {buildId}.",
+                                item.BuildId,
+                                item.ChannelId))
+                            {
+                                await RunAssociatedReleasePipelinesAsync(item.BuildId, item.ChannelId, cancellationToken);
+                            }
+                        }
+
+                        await tx.CommitAsync();
+                    }
+
+                    await Task.Delay(1000, cancellationToken);
```

Comment:
> await Task.Delay(1000, cancellationToken); [](start = 19, length = 43)

We probably only want to delay if we didn't find work to do. No reason to have the thing sitting around when there is a pile of stuff in the queue. :-)